cpp/autosar/test/rules/A13-3-1/FunctionThatContainsForwardingReferenceAsItsArgumentOverloaded.expected

```diff
@@ -3,3 +3,5 @@
 | test.cpp:50:3:50:3 | A | Function overloads a $@ with a forwarding reference parameter. | test.cpp:47:3:47:3 | A | function |
 | test.cpp:63:8:63:8 | B | Function overloads a $@ with a forwarding reference parameter. | test.cpp:66:3:66:3 | B | function |
 | test.cpp:63:8:63:8 | B | Function overloads a $@ with a forwarding reference parameter. | test.cpp:66:3:66:3 | B | function |
+| test.cpp:71:7:71:7 | C | Function overloads a $@ with a forwarding reference parameter. | test.cpp:74:25:74:25 | C | function |
```

Comment:
Why does the compliant example add rows?

Reply:
We want to exclude compiler generated functions such as the move and copy constructors that will match in parameter count.

Reply:
I currently disagree, but may be misunderstanding the rule description

on page 239 of the standard, in the examples for this rule - there is an explicit example (a constructor in class A) where the only reason that the explicit declared constructor is COMPLIANT is via exception, where the exception is due to the the template param constraints, and otherwise would be overloading a compiler generated move constructor

I think the testcase added still describes a situation where it could be unclear which function is called, which is the purpose of the rule, however willing to do a reconsider !

Reply:
Per discussion we want to alert on the override with the forward reference parameters, instead of the copy/move ctors, because it doesn't apply SFINAE for which we have an exception. This would align with the example in the standard.